src/net_processing.cpp

```diff
@@ -2991,14 +3011,14 @@ void ProcessMessage(
                 const auto current_time = GetTime<std::chrono::microseconds>();
 
                 if (!State(pfrom.GetId())->m_wtxid_relay) {
-                    for (const CTxIn& txin : tx.vin) {
+                    for (const uint256& parent_txid : unique_parents) {
                         // Here, we only have the txid (and not wtxid) of the
                         // inputs, so we only request parents from
                         // non-wtxid-relay peers.
                         // Eventually we should replace this with an improved
                         // protocol for getting all unconfirmed parents.
-                        CInv _inv(MSG_TX | nFetchFlags, txin.prevout.hash);
-                        pfrom.AddKnownTx(txin.prevout.hash);
+                        CInv _inv(MSG_TX | nFetchFlags, parent_txid);
```

Comment:
Since you're already touching this line, maybe we could remove `nFetchFlags` here.

`AlreadyHave()` and `RequestTx()` don't do anything differently depending on whether the CInv is an MSG_TX or MSG_WITNESSTX, so adding the `MSG_WITNESS_FLAG` is useless here. (the same is true for the `nFetchFlags` in the INV processing in ProcessMessages).

Perhaps best left for a different PR.

Reply:
Interesting point. If not done before, #19611 touches `nFetchFlags` while refactoring the INV processing in ProcessMessages (to take CInv::type private) and I can look at/update it there.

Reply:
> Since you're already touching this line, maybe we could remove `nFetchFlags` here.
> 
> `AlreadyHave()` and `RequestTx()` don't do anything differently depending on whether the CInv is an MSG_TX or MSG_WITNESSTX, so adding the `MSG_WITNESS_FLAG` is useless here. (the same is true for the `nFetchFlags` in the INV processing in ProcessMessages).
> 
> Perhaps best left for a different PR.

Thanks @jnewbery, am looking at removing these in #19611.

Reply:
Removed unused `nFetchFlags` from ProcessMessages TX and INV processing in #19611.

Reply:
Leaving this alone, thanks.